NavigationReactNative/src/android/app/src/main/java/com/navigation/reactnative/FragmentNavigator.java

```diff
@@ -17,10 +17,24 @@
 
 class FragmentNavigator extends SceneNavigator {
 
+    private SceneFragment getLastSceneFragment(FragmentManager fragmentManager){
+        int from = fragmentManager.getFragments().size() - 1;
+
+        for (int i = from; i >= 0; i--) {
+            Fragment current = fragmentManager.getFragments().get(i);
+
+            if (current instanceof  SceneFragment) {
+                return (SceneFragment) current;
+            }
+        }
+
+        return null;
+    }
+
     @Override
     void navigateBack(int currentCrumb, int crumb, Activity activity, NavigationStackView stack) {
         FragmentManager fragmentManager = getFragmentManager(stack, activity);;
-        SceneFragment fragment = (SceneFragment) fragmentManager.getFragments().get(fragmentManager.getFragments().size() - 1);
+        SceneFragment fragment = getLastSceneFragment(fragmentManager);
```

Comment:
We don't need the `getLastSceneFragment` method because all fragments are tagged with their place in the stack. So we can get the last fragment using this `oldKey` property which holds the tag of the current fragment.
```java
SceneFragment fragment = (SceneFragment) fragmentManager.findFragmentByTag(oldKey);
```
Can you test that, as well as fixing the original problem, the change works in the zoom sample because its shared element transitions rely on this line of code. There are 2 test cases to check:
1. Select a color, press back and check the shared element transition runs
2. Select a color, then change the color on the details page, press back and check the shared element transition runs.

Reply:
Thanks for the comment.

I will check it and let you know